File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java
package com.mentalmachines.droidcon_boston.firebase;

import android.util.Log;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.mentalmachines.droidcon_boston.R;
import com.mentalmachines.droidcon_boston.utils.NotificationUtils;
import java.util.Map;

public class DbFirebaseMessagingService extends FirebaseMessagingService {

    private final String TAG = getClass().getSimpleName();

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {

        if (remoteMessage != null) {
            Map<String, String> payloadMap = remoteMessage.getData();
            String bodyStr = "NA";

            // Check if message contains a data payload.
            if (payloadMap.size() > 0) {
                Log.d(TAG, "Payload: ");
                for (String key : payloadMap.keySet()) {
                    Log.d(TAG, "Key: " + key + ", Value: " + payloadMap.get(key));
                }
            }

            // Check if message contains a notification payload.
            if (remoteMessage.getNotification() != null) {
                bodyStr = remoteMessage.getNotification().getBody();
                Log.d(TAG, "Body: " + bodyStr);
            }

            // Show the notification here
            NotificationUtils notificationUtils = new NotificationUtils(this);
            notificationUtils.sendAndroidChannelNotification(getString(R.string.conference_name), bodyStr, 101);
        }


    }
}

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.kt
package com.mentalmachines.droidcon_boston.firebase

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.mentalmachines.droidcon_boston.R
import com.mentalmachines.droidcon_boston.utils.NotificationUtils

class DbFirebaseMessagingService : FirebaseMessagingService() {

    private val TAG = javaClass.simpleName

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {
        if (remoteMessage != null) {
            val payloadMap = remoteMessage.data
            var bodyStr: String? = "NA"

            // Check if message contains a data payload.
            if (payloadMap.isNotEmpty()) {
                Log.d(TAG, "Payload: ")
                for (key in payloadMap.keys) {
                    Log.d(TAG, "Key: " + key + ", Value: " + payloadMap[key])
                }
            }

            // Check if message contains a notification payload.
            if (remoteMessage.notification != null) {
                bodyStr = remoteMessage.notification!!.body
                Log.d(TAG, "Body: " + bodyStr!!)
            }

            // Show the notification here
            val notificationUtils = NotificationUtils(this)
            notificationUtils.sendAndroidChannelNotification(getString(R.string.conference_name), bodyStr!!, 101)
        }
    }
}
